File: Tree/tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def construct_tree(o_list):
    if o_list is None or o_list[0] is None:
        return None
    i=1

    root = TreeNode(o_list[0])
    queue = [root]
    while i<len(o_list):
        cur = queue.pop(0)
        if o_list[i] is not None:
            cur.left = TreeNode(o_list[i])
            queue.append(cur.left)
        i+=1
        if o_list[i] is not None and i<len(o_list):
            cur.right = TreeNode(o_list[i])
            queue.append(cur.right)
        i+=1

    return root

if __name__ == '__main__':
    # To construct a tree, we need a explicit list
    # for example, [3,9,20,null,null,15,7]
    # Tree level traversal
    order_list = [3,9,20,None,None,15,7]
    tree = construct_tree(order_list)

File: Tree/tree.cpp
#include "tree.h"

TreeNode* ConstructTree(const vector<int>& order)
{
    if (order.empty() || order[0] == INT_MIN) return nullptr;

    queue<TreeNode*> q;
    TreeNode* root = new TreeNode(order[0]);
    q.push(root);
    int i = 1;

    while (!q.empty() && i < order.size()) {
        TreeNode* current = q.front();
        q.pop();

        if (i < order.size() && order[i] != INT_MIN) {
            current->left = new TreeNode(order[i]);
            q.push(current->left);
        }
        i++;

        if (i < order.size() && order[i] != INT_MIN) {
            current->right = new TreeNode(order[i]);
            q.push(current->right);
        }
        i++;
    }

    return root;
}

int main()
{
    vector<int> list_1 = {3,9,20,NULL,NULL,15,7};
    vector<int> list_2 = {1,2,2,3,4,4,3};
    TreeNode *p = ConstructTree(list_1);
    TreeNode *q = ConstructTree(list_2);

    Tree_ops ops;
    cout << "Max depth of tree p is: " << ops.maxDepth(p) << endl;
}